src/acanban/auth.py

```diff
@@ -54,40 +56,61 @@ class User(AuthUser):
 
     def __init__(self, username: Optional[str]) -> None:
         super().__init__(username)
-        password, name, email, role = current_app.auth_manager.users[username]
         self.username = username
-        self.password = password
-        self.name = name
-        self.email = email
-        self.role = role
+        self.r = r.table('users').get(self.username)
+
+    @property
+    async def password(self) -> str:
+        async with current_app.db_pool.connection() as connection:
+            return await self.r['password-hash'].run(connection)
+
+    @property
+    async def name(self) -> str:
+        async with current_app.db_pool.connection() as connection:
+            return await self.r['name'].run(connection)
+
+    @property
+    async def email(self) -> str:
+        async with current_app.db_pool.connection() as connection:
+            return await self.r['email'].run(connection)
+
+    @property
+    async def role(self) -> str:
+        async with current_app.db_pool.connection() as connection:
+            return await self.r['role'].run(connection)
 
 
 class Authenticator(AuthManager):
-    users: Dict[Optional[str], Tuple[Optional[str], Optional[str],
-                                     Optional[str], Optional[str]]]
     user_class = User
 
     def init_app(self, app: Quart) -> None:
         """Embed auth_manager attribute into app."""
         super().init_app(app)
-        # TODO: wrap the database instead
-        self.users = {None: (None, None, None, None)}
 
-    def add_user(self, username: str, password: str,
-                 name: str, email: str, role: str) -> None:
+    async def add_user(self, username: str, password: str,
+                       name: str, email: str, role: str) -> None:
         """Try to add user to database."""
-        if username in self.users: raise ValueError('username taken')
         # This never happens through browser, but via manual requests.
         if role not in ROLES: raise ValueError('unknown role')
-        self.users[username] = crypt(password), name, email, role
-
-    def log_user(self, username: str, password: str) -> User:
+        async with current_app.db_pool.connection() as connection:
+            user = await r.table('users').get(username).run(connection)
+            if user is not None:
+                raise ValueError('username taken')
+            await r.table('users').insert({'username': username,
+                                           'password-hash': crypt(password),
+                                           'name': name, 'email': email,
+                                           'role': role}).run(connection)
```

Comment:
[By default conflict will be treated as error](https://rethinkdb.com/api/python/insert/) (*Duplicate primary key*, regardless if other fields is the same):

```suggestion
            user = {'username': username, 'password-hash': crypt(password),
                    'name': name, 'email': email, 'role': role}
            if await self.r.insert(user).run(connection)['errors']:
                raise ValueError('username taken')
```